File: typescript/src/models/paymentData.ts
// this file is @generated
import { type Address, AddressSerializer } from "./address";
import {
  type PaymentProviderType,
  PaymentProviderTypeSerializer,
} from "./paymentProviderType";

export interface PaymentData {
  token: string;
  provider: PaymentProviderType;
  billingAddress?: Address | null;
}

export const PaymentDataSerializer = {
  _fromJsonObject(object: any): PaymentData {
    return {
      token: object["token"],
      provider: PaymentProviderTypeSerializer._fromJsonObject(object["provider"]),
      billingAddress: object["billing_address"]
        ? AddressSerializer._fromJsonObject(object["billing_address"])
        : undefined,
    };
  },

  _toJsonObject(self: PaymentData): any {
    return {
      token: self.token,
      provider: PaymentProviderTypeSerializer._toJsonObject(self.provider),
      billing_address: self.billingAddress
        ? AddressSerializer._toJsonObject(self.billingAddress)
        : undefined,
    };
  },
};

File: rust/src/models/payment_data.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{address::Address, payment_provider_type::PaymentProviderType};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct PaymentData {
    pub token: String,
    pub provider: PaymentProviderType,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub billing_address: Option<Address>,
}
